### PR TITLE
don't error when a channel doesn't have any community posts

### PR DESCRIPTION
**Pull request by ChunkyProgrammer**
we now return an empty array when there are no posts.

closes https://github.com/iv-org/invidious/issues/3814
https://github.com/FreeTubeApp/FreeTube/issues/3499

### CODE REVIEW

**Review by syeopite — APPROVED** [review]

**Comment by Fijxu** [comment]
I wanted to test this but I get "This channel hasn't posted yet" even if I don't have this pull request merged into my debug instance and that string doesn't even exists on the Invidious code.

image


image


image

However if I merge this pull request into my debug instance I don't even get the message This channel hasn't posted yet on Invidious. What is going on?

image

**Comment by syeopite** [comment]
That's the point. This PR makes Invidious response to blank responses from YouTube the same by displaying an empty page instead of an error message. 

I'm not sure if I agree with not showing a message though. 
https://github.com/iv-org/invidious/issues/3814#issuecomment-1918129165

**Comment by ChunkyProgrammer** [comment]
> I'm not sure if I agree with not showing a message though. [#3814 (comment)](https://github.com/iv-org/invidious/issues/3814#issuecomment-1918129165)

I can add a message to the invidious ui for the empty tabs (that way we can also include it in weblate). I don't think it would be that beneficial to include that message in the api imo

File: src/invidious/channels/community.cr [hunk 1]
```diff
@@ -57,6 +57,17 @@ def extract_channel_community(items, *, ucid, locale, format, thin_mode, is_sing
       .try &.as_s || ""
     if error_message == "This channel does not exist."
       raise NotFoundException.new(error_message)
+    elsif error_message == "This channel hasn't posted yet"
+      response = JSON.build do |json|
+        json.object do
+          json.field "authorId", ucid
+          json.field "comments" do
+            json.array do
+            end
```
Comment (SamantazFox) [inline src/invidious/channels/community.cr:66]:
```suggestion
            json.array {}
```

File: src/invidious/channels/community.cr [hunk 2]
```diff
@@ -57,6 +57,17 @@ def extract_channel_community(items, *, ucid, locale, format, thin_mode, is_sing
...
+          end
```
Comment (SamantazFox) [inline src/invidious/channels/community.cr:67]:
```suggestion
          json.field "comments", json.array {}
```

**Comment by syeopite** [comment]
> > I'm not sure if I agree with not showing a message though. [#3814 (comment)](https://github.com/iv-org/invidious/issues/3814#issuecomment-1918129165)
> 
> I can add a message to the invidious ui for the empty tabs (that way we can also include it in weblate). I don't think it would be that beneficial to include that message in the api imo

Yes please, that'd be great. This PR just comes with an unintentional side-effect of also removing the error message on the frontend side of things.

**Comment by ChunkyProgrammer** [comment]
When a community tab is empty it will display the message in the UI now :smile:

**Review by SamantazFox — APPROVED** [review]

**Comment by ChunkyProgrammer** [comment]
build failure is unrelated to this PR and caused by changes in crystal nightly

**Review by SamantazFox — APPROVED** [review]

**Comment by ChunkyProgrammer** [comment]
YouTube seems to be hiding these channels now

**Comment by ChunkyProgrammer** [comment]
Yeah, this PR shouldn't be needed now.

**Pull request closed without merging**